### DISCUSSION TITLE
MTU Path Discovery and Explicit Congestion Notification (ECN)

### DISCUSSION THREAD

**Discussion by nlifew** [libuv/libuv#4487] [General]
Hi,

I am planning to use libuv as a cross-platform networking library and integrate it with some existing QUIC open-source libraries. However, I encountered a problem: I cannot precisely control the IP header information, which makes it impossible to know anything about MTU Path Discovery and Explicit Congestion Notification (ECN).

Back in 2020, a diligent engineer (@jasnell ) already raised this issue(#3077 ), but it's been four years with no follow-up(also possible that I may have missed any updates). I would like to know if the community has any thoughts or plans regarding this matter?

Thank you!

**Comment by santigimeno** [comment]
I'm not aware of any work being done on this topic. I think https://github.com/libuv/libuv/issues/3077#issuecomment-751680889 still stands as a valid stance for any new functionality that may end in libuv.

**Comment by jasnell** [comment]
I certainly have not had the time to pursue this but would very much like to see someone pick this up!
